### DISCUSSION TITLE
4.38 Beta3 Feedback | Passkey popup?

### DISCUSSION THREAD

**Discussion by karlis-vagalis** [authelia/authelia#6328] [Pre-Release Feedback]
Hello,

thanks for the great work for the upcoming release. I decided to try out the beta on my own homelab and here is some feedback or stuff I have noticed this far:

1. Using Firefox with Bitwarden extension enabled. When managing user's devices and trying to add a new `WebAuthn Credentials`, a pop-up appears, which offers to save the passkey for this site to your Bitwarden vault (for a specific login entry). Is this behavior correct? Does Authelia v4.38 introduce support for passkeys? Secondly, going through with it, the new "device" is successfully added, but when you try to log in using it as second factor, it fails with the following error:
```
level=error
method=POST
msg=Unable to parse WebAuthn authentication challenge for user 'x-redacted-x': Parse error for Assertion
path=/api/secondfactor/webauthn
remote_ip=192.168.2.26
stack=[
    {
        "File": "github.com/authelia/authelia/v4/internal/handlers/response.go",
        "Line": 303,
        "Name": "respondUnauthorized"
    },
    {
        "File": "github.com/authelia/authelia/v4/internal/handlers/handler_sign_webauthn.go",
        "Line": 146,
        "Name": "WebAuthnAssertionPOST"
    },
    {
        "File": "github.com/authelia/authelia/v4/internal/middlewares/require_auth.go",
        "Line": 19,
        "Name": "Require1FA.func1"
    },
    {
        "File": "github.com/authelia/authelia/v4/internal/middlewares/bridge.go",
        "Line": 54,
        "Name": "handleRouter.(*BridgeBuilder).Build.func6.1"
    },
    {
        "File": "github.com/authelia/authelia/v4/internal/middlewares/headers.go",
        "Line": 56,
        "Name": "SecurityHeadersCSPNone.func1"
    },
    {
        "File": "github.com/authelia/authelia/v4/internal/middlewares/headers.go",
        "Line": 82,
        "Name": "SecurityHeadersNoStore.func1"
    },
    {
        "File": "github.com/authelia/authelia/v4/internal/middlewares/headers.go",
        "Line": 12,
        "Name": "SecurityHeaders.func1"
    },
    {
        "File": "github.com/fasthttp/router@v1.4.22/router.go",
        "Line": 420,
        "Name": "(*Router).Handler"
    },
    {
        "File": "github.com/authelia/authelia/v4/internal/middlewares/log_request.go",
        "Line": 14,
        "Name": "handleRouter.LogRequest.func40"
    },
    {
        "File": "github.com/authelia/authelia/v4/internal/middlewares/strip_path.go",
        "Line": 22,
        "Name": "handleRouter.StripPath.func41.1"
    },
    {
        "File": "github.com/valyala/fasthttp@v1.51.0/server.go",
        "Line": 2359,
        "Name": "(*Server).serveConn"
    },
    {
        "File": "github.com/valyala/fasthttp@v1.51.0/workerpool.go",
        "Line": 224,
        "Name": "(*workerPool).workerFunc"
    },
    {
        "File": "github.com/valyala/fasthttp@v1.51.0/workerpool.go",
        "Line": 196,
        "Name": "(*workerPool).getCh.func1"
    },
    {
        "File": "runtime/asm_amd64.s",
        "Line": 1650,
        "Name": "goexit"
    }
]
```

2. Secondly, if you have aforementioned Bitwarden extension enabled, the prompt from the extension always pops up when trying to add new WebAuthn device, which makes you either need to disable the extension or close pop-up. However, closing introduces another error, where afterwards, you are not able to add new device like Yubikey. It simply does not prompt for device PIN anymore. One has to reload the page completely to start over.

**Comment by james-d-elliott** [comment]
We do not specifically support passkeys or intend to at this stage. That's why it's not mentioned at all in any of the docs or articles posted.

As far as the issues surrounding extensions and their hijacking of the WebAuthn API see the discussion here: https://github.com/w3c/webauthn/issues/1976

In short they're monkey patching the browser and this leads to quite a few issues which can lead to strange issues like this.

In regards to the inability to register (no PIN etc) I can't really replicate this. Is there a way in a supported scenario to do so?

**Reply by karlis-vagalis** [reply]
Thanks for the info, good to know.

Maybe the second one is indeed specific to my browser/config. I mean, it did work eventually. Thanks anyway!

**Reply by james-d-elliott** [reply]
I'll see if I can replicate it with some more normal flows.. it is a bit odd that Bitwarden isn't respecting the fact we're deliberately asking not to be provided a passkey. But I did manage to replicate some of this.

**Reply by james-d-elliott** [reply]
So it looks like the credentials generated by Bitwarden are generally not to standard. They are backed up to the vault but they don't have the BE/BS flags set correctly or are not reporting them correctly. Overall very odd.

**Comment by karlis-vagalis** [comment]
Also something that I noticed. Done on mobile device.

In the "Manage my devices" section, the TOTP section is enabled an theoretically allows me to add TOTP, however, this option as second factor is disabled in the configuration file. Would it be possible to disable/grey out that TOTP section in general in that case? So it is not misleading? When trying to add TOTP while it was not allowed in configuration file just makes it stuck on "loading" in the first step.

**Reply by james-d-elliott** [reply]
Nice spot. But please make a separate topic as it's unrelated.
